### PR TITLE
OPTY-393: Beefy invest limit

### PR DESCRIPTION
**Pull request by mariogutval**
- Added investLimit to Beefy
- Removed isContract calls
- Bumped up solidity version in Beefy adapter

#### PR Checklist

- [ ] Tests

### CODE REVIEW

File: contracts/2_matic/beefy.finance/BeefyFinanceAdapter.sol [hunk 1]
```diff
@@ -31,10 +26,9 @@ import { IWETH } from "@uniswap/v2-periphery/contracts/interfaces/IWETH.sol";
 contract BeefyFinanceAdapter is
     IAdapter,
     IAdapterHarvestReward,
-    IAdapterStaking //REPLACE IAdapterHarvest..
+    IAdapterStaking, //REPLACE IAdapterHarvest..
```
Comment (dhruvinparikh) [inline contracts/2_matic/beefy.finance/BeefyFinanceAdapter.sol]: - [ ] remove this comment

File: contracts/1_ethereum/lido.fi/LidoAdapter.sol [hunk 2]
```diff
@@ -59,7 +59,6 @@ contract LidoAdapter is IAdapter, IAdapterInvestLimit, AdapterModifiersBase {
         override
         onlyRiskOperator
     {
-        require(_underlyingToken.isContract(), "!isContract");
         maxDepositPoolPct[_underlyingToken] = _maxDepositPoolPct;
```
Comment (dhruvinparikh) [inline contracts/1_ethereum/lido.fi/LidoAdapter.sol]: - [ ] please revert any changes made to lido as it is already deployed. We can curate in adapter refactoring document as what changes needs to be made.

**Review by dhruvinparikh — CHANGES_REQUESTED** [review]

File: contracts/2_matic/beefy.finance/BeefyFinanceAdapter.sol [hunk 3]
```diff
@@ -260,17 +253,23 @@ contract BeefyFinanceAdapter is
...
+            _underlyingToken,
+            _amount,
+            getPoolValue(_liquidityPool, address(0))
+        );
+        if (_depositAmount > 0) {
             _codes = new bytes[](3);
             _codes[0] = abi.encode(
                 _underlyingToken,
                 abi.encodeWithSignature("approve(address,uint256)", _liquidityPool, uint256(0))
             ); //maybe there are two pools in case in some cases we wanted to deposit two tokens?
```
Comment (deepcrazy) [inline contracts/2_matic/beefy.finance/BeefyFinanceAdapter.sol]: I think we can remove this comment.

File: contracts/2_matic/beefy.finance/BeefyFinanceAdapter.sol [hunk 4]
```diff
@@ -260,17 +253,23 @@ contract BeefyFinanceAdapter is
...
             _codes[1] = abi.encode(
                 _underlyingToken,
-                abi.encodeWithSignature("approve(address,uint256)", _liquidityPool, _amount)
+                abi.encodeWithSignature("approve(address,uint256)", _liquidityPool, _depositAmount)
             );
-            _codes[2] = abi.encode(_liquidityPool, abi.encodeWithSignature("deposit(uint256)", _amount));
+            _codes[2] = abi.encode(_liquidityPool, abi.encodeWithSignature("deposit(uint256)", _depositAmount));
         }
```
Comment (deepcrazy) [inline contracts/2_matic/beefy.finance/BeefyFinanceAdapter.sol]: Isn't following better? I think, we should check first if `_amount > 0` or not. and later check `_depositAmount > 0` or not. If =0, then revert the transaction.
```
  if (_amount > 0) {
      uint256 _depositAmount = _getDepositAmount(
              _liquidityPool,
              _underlyingToken,
              _amount,
              getPoolValue(_liquidityPool, address(0))
          );
        
        if (_depositAmount > 0) {
            _codes = new bytes[](3);
            _codes[0] = abi.encode(
                _underlyingToken,
                abi.encodeWithSignature("approve(address,uint256)", _liquidityPool, uint256(0))
            ); 
            _codes[1] = abi.encode(
                _underlyingToken,
                abi.encodeWithSignature("approve(address,uint256)", _liquidityPool, _depositAmount)
            );
            _codes[2] = abi.encode(_liquidityPool, abi.encodeWithSignature("deposit(uint256)", _depositAmount));
        }
  }
```

Thoughts?

**Review by deepcrazy — CHANGES_REQUESTED** [review]

**Review by dhruvinparikh — APPROVED** [review]

**Review by deepcrazy — APPROVED** [review]

**Pull request merged**